core/shared/src/main/scala/scalaz/zio/Promise.scala

```diff
@@ -90,7 +89,7 @@ class Promise[E, A] private (private val state: AtomicReference[State[E, A]]) ex
    * Interrupts the promise with the specified list of throwable(s). This will interrupt
    * all fibers waiting on the value of the promise.
    */
-  final def interrupt0(ts: List[Throwable]): IO[Nothing, Boolean] = done(ExitResult.Terminated[E, A](ts))
+  final def interrupt0(ts: List[Throwable]): IO[Nothing, Boolean] = done(ExitResult.interrupted(ts))
```

Comment:
There is only need for 1 interrupt method:

```scala
def interrupt: IO[Nothing, Boolean]
```

i.e. we can get rid of the list of throwables.